### PR TITLE
(WIP) Initial draft for the rest api doc

### PR DESCRIPTION
**Pull request by Ferdi**
#118

See http://docs.lore.apiary.io/

### CODE REVIEW

File: apiary.apib [hunk 1]
```diff
@@ -0,0 +1,333 @@
...
+
+# URL
+URL Structure is `https://{domain}/api/v1/{resource}/{resourceId}`
+
+| Code   | Description |
+| ---    | --- |
+|`{domain}`    | The domain of the LORE service. `private-b0103d-lore.apiary-mock.com/` is the Apiary mock server. |
+|`/api/v1/`    | The service endpoint includes the current version (v1) |
+|`{resource}`  | The resource endpoint for a specific set of objects in the system. Supported endpoints include repositories, vocabularies, learningresources, members|
+|`{resource_idd}`  |The `resource_id` sets the ID for a specific object to reference |
```
Comment (pdpinch) [inline apiary.apib]: typo: `{resource_id}` instead of `{resource_idd}`

File: apiary.apib [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+### Create a New Repository [POST]
+
+Create a repository by providing its JSON representation
+
++ Request (application/json)
+
+        {
+            "name": "Physics 1",
+            "description": "All intro-to-physics courses"
+        }
```
Comment (pdpinch) [inline apiary.apib]: should we return the slug also?

File: apiary.apib [hunk 3]
```diff
@@ -0,0 +1,333 @@
...
+    + Headers
+
+            Location: /repos/123
+
+    + Body
+
+            {
+                "name": "Physics 1",
+                "description": "All intro-to-physics courses"
+            }
```
Comment (pdpinch) [inline apiary.apib:120]: Should we return the slug also?

File: apiary.apib [hunk 4]
```diff
@@ -0,0 +1,333 @@
...
+|`500`| Server error. |
+
+
+# Authentication
+
+tbd.
+
+# Pagination
+
+Requests that return multiple items all accept a `?page` parameter
```
Comment (noisecapella) [inline apiary.apib:57]: How many items per page? Default is 10 for django-rest-framework

File: apiary.apib [hunk 5]
```diff
@@ -0,0 +1,333 @@
...
+Page numbering is 1-based. Omitting the ?page parameter will return the first page.
+
+# LORE API Root [/]
+
+## Retrieve the Entry Point [GET]
+
++ Response 200 (application/json)
+
+        {
+            "repositories_url": "/repos"
```
Comment (noisecapella) [inline apiary.apib:70]: Providing links like this is a property of HATEOAS, is that what we're going for?

File: apiary.apib (hunk 3 above)
Comment (noisecapella) [inline apiary.apib:120]: I'm not sure it's relevant for a REST api since we're using id numbers in the URL

File: apiary.apib (hunk 4 above)
Comment (Ferdi) [inline apiary.apib:57]: I suggest  ~ 20  for our case

File: apiary.apib [hunk 6]
```diff
@@ -0,0 +1,333 @@
...
++ Response 201 (application/json)
+
+    + Headers
+
+            Location: /repos/123
+
+    + Body
+
+            {
+                "name": "Physics 1",
```
Comment (noisecapella) [inline apiary.apib]: Should include an id number to reference it in the future. I'm not sure that the Location header is useful here

File: apiary.apib [hunk 7]
```diff
@@ -0,0 +1,333 @@
...
+
++ Parameters
+    + repo_id: `17` (number, required) - ID of the Repository
+
+### Retrieve a repository [GET]
+
++ Response 200 (application/json)
+
+        {
+            "tbd": "tbd",
```
Comment (noisecapella) [inline apiary.apib:89]: This should be the same as the return from 'Create a new repository'

File: apiary.apib [hunk 8]
```diff
@@ -0,0 +1,333 @@
...
+            "repositories_url": "/repos"
+        }
+
+
+
+## Group Repositories
+
+Repository resource
+
+## Repositories [/repos/{repo_id}]
```
Comment (noisecapella) [inline apiary.apib]: Just to confirm, we're using /repos/ here instead of /repositories/?

File: apiary.apib [hunk 9]
```diff
@@ -0,0 +1,333 @@
...
+                {
+                    "tbd": "tbd",
+                }
+            ]
+
+## Group Learningobjects
+
+Learning Object resource
+
+## Learningobjects [/repos/{repo_id}/learningobjects/{learningobject_id}]
```
Comment (noisecapella) [inline apiary.apib:145]: Should be learningresources

**Comment by Ferdi** [comment]
2 questions for discussions so far:
- Should we use `{id}` or `{slug}`
- Should the name in the url be "repos" or "repositories"

I dont have a strong opinion on either.

**Comment by carsongee** [comment]
I'm a slug fan ala github API (since it is my favorite so far), also github uses repos, but there docs say repositories, but I don't care much on that front

**Comment by carsongee** [comment]
https://developer.github.com/v3/repos/#list-your-repositories

File: apiary.apib (hunk 6 above)
Comment (Ferdi) [inline apiary.apib]: I think Location header is a sort of requirement on "POST"  http://goo.gl/87Mti0 .  I agree that we should return ID or the full representation of the newly created resource.

**Comment by Ferdi** [comment]
Good idea on following github design when we can. It's one of my favorite apis too.

**Comment by pdpinch** [comment]
@noisecapella what else needs to be addressed before we merge this?

**Comment by noisecapella** [comment]
@pdpinch I haven't done any work on this yet, I've changed it back to work in progress

**Pull request merged**